ts/src/profiler.ts

```diff
@@ -125,6 +141,50 @@ async function profileBytes(p: perftools.profiles.IProfile): Promise<string> {
   return gzBuf.toString('base64');
 }
 
+/**
+ * Error constructed from http server response which indicates backoff.
+ */
+class BackoffResponseError extends Error {
+  backoffMillis: number;
+  constructor(response: ServerBackoffResponse) {
+    super(response.statusMessage);
+    this.backoffMillis = parseDuration(response.body.details.retryDelay);
+  }
+}
+
+/**
+ * @return - true if error is a BackoffResponseError and false otherwise
+ */
+function isBackoffResponseError(err: Error): err is BackoffResponseError {
+  // tslint:disable-next-line: no-any
+  return typeof (err as any).backoffMillis === 'number';
+}
+
+/**
+ * Class which tracks how long to wait before the next retry and can be
+ * used to get this backoff.
+ */
+export class Retryer {
+  private nextMaxBackoffMillis: number;
+  constructor(
+      readonly initialBackoffMillis: number, readonly maxBackoffMillis: number,
+      readonly backoffMultiplier: number) {
+    this.nextMaxBackoffMillis = this.initialBackoffMillis;
+  }
+  getBackoff(): number {
+    const curBackoff = Math.random() * this.nextMaxBackoffMillis;
+    this.nextMaxBackoffMillis =
+        this.backoffMultiplier * this.nextMaxBackoffMillis;
+    if (this.nextMaxBackoffMillis > this.maxBackoffMillis) {
```

Comment:
Just `this.nextBackoffMillis = Math.max(this.nextBackoffMillis, this.maxBackoffMillis);`

Reply:
done.